### PR TITLE
Sovereign cloud support added

### PR DESCRIPTION
**Pull request by tauhid621**
Instead of hardcoded management url we will use "az cloud show" command to get the management url

### CODE REVIEW

File: src/azure/managementUrlHelper.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import * as core from '@actions/core';
+import * as exec from '@actions/exec';
+import * as io from "@actions/io";
+
+export class ManagementUrlHelper{
+
+    public static async getBaseUrl(): Promise<string> {
+        if(!this._baseUrl) {            
+            let azCloudDetails = JSON.parse(await ManagementUrlHelper.executeAzCliCommand('cloud show'));
```
Comment (ajinkya599) [inline src/azure/managementUrlHelper.ts:9]: Should we wrap the baseUrl derivation logic in `try..catch`, just to be safe?

File: src/azure/managementUrlHelper.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+            const cloudEndpoints = azCloudDetails['endpoints'];
+            this._baseUrl = ManagementUrlHelper.getResourceManagerUrl(cloudEndpoints);
+        }
+
+        return this._baseUrl;
+    }
+    
+    public static async executeAzCliCommand(command: string, args?: string[]): Promise<string> {
```
Comment (ajinkya599) [inline src/azure/managementUrlHelper.ts:17]: This method looks duplicate of `getTokenFromAzCLI()` in `azAuthentication`. Suggestion:
- Make this class `AzCli`
- Make this method `executeCommand`
- Remove `getTokenFromAzCLI` from `azAuthentication` and use `AzCli.azAuthentication()`

**Review by ajinkya599 — APPROVED** [review]

File: src/azure/azHttpClient.ts [hunk 3]
```diff
@@ -242,7 +242,7 @@ export class AzHttpClient {
   }
 
   private getRoleAssignmentUrl(scope: string, roleAssignmentId: string): string {
-    return `https://management.azure.com${scope}/providers/Microsoft.Authorization/roleAssignments/${roleAssignmentId}?api-version=${this.roleApiVersion}`
+    return `${this.managementUrl}${scope}/providers/Microsoft.Authorization/roleAssignments/${roleAssignmentId}?api-version=${this.roleApiVersion}`
```
Comment (ajinkya599) [inline src/azure/azHttpClient.ts:245]: Not sure whether all environments support the API versions being used for the different APIs. Good to test E2E!

File: src/azure/managementUrlHelper.ts (hunk 2 above)
Comment (tauhid621) [reply src/azure/managementUrlHelper.ts]: Created AzCli Class. Moved getAccessToken method here as well

File: src/azure/managementUrlHelper.ts (hunk 1 above)
Comment (tauhid621) [reply src/azure/managementUrlHelper.ts]: try catch added.

File: src/azure/azHttpClient.ts (hunk 3 above)
Comment (tauhid621) [reply src/azure/azHttpClient.ts:244]: Will need to test it properly

**Pull request merged**